### DISCUSSION TITLE
Problems installing on Arudino Nano ESP32

### DISCUSSION THREAD

**Discussion by Bel1haz0r** [PlummersSoftwareLLC/NightDriverStrip#639] [Q&A]
Did anyone install this on Arduino Nano ESP32? I first tried with webinstall but every time it would fail with this error:
Failed to initialize. Try resetting your device or holding the BOOT button while clicking INSTALL.

I also tried to clone the repo and build it from VSCode but there I am having an anoyance with node:

site\node_modules\vite\bin\vite.js:2
import { performance } from 'node:perf_hooks'
           ^

SyntaxError: Unexpected token {

I must admit I am not that familiar with node although I use it for various projects I never really dug deeper into it due to lack of free time.

Anyhelp is appreciated.
Cheers

**Comment by robertlipe** [comment]
I don't think it's come up before, but we can probably figure it out
together.

Looking at
https://store-usa.arduino.cc/products/nano-esp32?selectedStore=us
it looks like this is an ESP32-S3, so you may be able to use one of the
targets for our existing s3 boards (look in platform.ini) I usually start
with lilygo-tdisplay-s3 and make my own custom one for my S3 boards. Here's
an example from one of my S3 boards (which I should clean up and submit):

➜  nightdriverstrip git:(s3-work) ✗ cat custom_yd.ini
[dev_yd]
extends         = dev_esp32-s3
board           = esp32-s3-devkitc-1
board_upload.flash_size = 8MB
board_build.partitions = default.csv

[env:yd-demo]
[env:yd-esp32-s3-demo]
extends         = dev_yd
build_flags     = -DDEMO=1
; -Os
; -ffunction-sections
; -fdata-sections
                  -DENABLE_WIFI=0
                  -DINCOMING_WIFI_ENABLED=0
                  -DENABLE_NTP=0
                  -DENABLE_AUDIO=0
                  -DUSE_PSRAM=0
                  ${dev_esp32-s3.build_flags}
lib_ignore      = U8g2
                  Bounce2
                  QRCode
                  IRremoteESP8266
                  ; TJpg_Decoder
;platform_packages =
;             framework-espidf@
https://github.com/espressif/esp-idf.git#release/v5.2
;
; platformio/framework-arduinoespressif32-libs @
https://github.com/espressif/esp32-arduino-libs.git#idf-release/v5.1

After that, I can build environment yd-esp32-s3-demo which runs. It sprays
rainbows out pin 5 and has remote on pin 4 and so on. If I wanted different
effects, I'd replace the -DDEMO with one of the other effects definitions
(which aren't clearly lableed, sigh) in i*/globals.ini..

Just make a custom_adafruit_nano_esp32.ini that has the settings you like.
Once you're satisfied with it, send it to the list in a pull request and
hopefully it can become part of the project for the next one in your shoes.

I haven't seen the error you're describing, but it looks freakishly like a
submit that went in this morning. Try rolling the tree back a day and see
if it buildds. It may be that the automated thingy that sent us the update
requires yesterday's version of some tool that might not yet be in common
use. Thiat's web development for you.

Good luck. Keep us posted.
RJL
